src/zimcheck.cpp

```diff
@@ -140,7 +160,7 @@ std::vector<std::string> getLinks(const std::string& page, bool withHref = true)
         // [TODO] Handle escape char
         while(*p != delimiter)
             p++;
-        links.push_back(std::string(linkStart, p));
+        links.push_back(removeLocalTagAndParameters(std::string(linkStart, p)));
```

Comment:
In fact, we may not need the function `removeLocalTagAndParameters` at all.
We could simply end our link when p is equal to delimiter or `#` or `?` (line 161)

The function has the advantage to be reusable, but we are not reusing it and, with the function, we create a too long string to truncate it just after.

Reply:
Yes, even if the problem is a bit more complex than this. Will propose something smarter. Thx for review.